bot/cogs/moderation/silence.py

```diff
@@ -0,0 +1,175 @@
+from typing import Optional
+
+from discord import PermissionOverwrite, TextChannel
+from discord.ext import commands, tasks
+from discord.ext.commands import Context
+
+from bot.bot import Bot
+from bot.constants import Channels, Emojis, Guild, MODERATION_ROLES, Roles
+from bot.converters import HushDurationConverter
+from bot.utils.checks import with_role_check
+
+log = logging.getLogger(__name__)
+
+
+class FirstHash(tuple):
+    """Tuple with only first item used for hash and eq."""
+
+    def __new__(cls, *args):
+        """Construct tuple from `args`."""
+        return super().__new__(cls, args)
+
+    def __hash__(self):
+        return hash((self[0],))
+
+    def __eq__(self, other: "FirstHash"):
+        return self[0] == other[0]
+
+
+class SilenceNotifier(tasks.Loop):
+    """Loop notifier for posting notices to `alert_channel` containing added channels."""
+
+    def __init__(self, alert_channel: TextChannel):
+        super().__init__(self._notifier, seconds=1, minutes=0, hours=0, count=None, reconnect=True, loop=None)
+        self._silenced_channels = set()
+        self._alert_channel = alert_channel
+
+    def add_channel(self, channel: TextChannel) -> None:
+        """Add channel to `_silenced_channels` and start loop if not launched."""
+        if not self._silenced_channels:
+            self.start()
+            log.info("Starting notifier loop.")
+        self._silenced_channels.add(FirstHash(channel, self._current_loop))
+
+    def remove_channel(self, channel: TextChannel) -> None:
+        """Remove channel from `_silenced_channels` and stop loop if no channels remain."""
+        with suppress(KeyError):
+            self._silenced_channels.remove(FirstHash(channel))
+            if not self._silenced_channels:
+                self.stop()
+                log.info("Stopping notifier loop.")
+
+    async def _notifier(self) -> None:
+        """Post notice of `_silenced_channels` with their silenced duration to `_alert_channel` periodically."""
+        # Wait for 15 minutes between notices with pause at start of loop.
+        if self._current_loop and not self._current_loop/60 % 15:
+            log.debug(
+                f"Sending notice with channels: "
+                f"{', '.join(f'#{channel} ({channel.id})' for channel, _ in self._silenced_channels)}."
+            )
+            channels_text = ', '.join(
+                f"{channel.mention} for {(self._current_loop-start)//60} min"
+                for channel, start in self._silenced_channels
+            )
+            await self._alert_channel.send(f"<@&{Roles.moderators}> currently silenced channels: {channels_text}")
+
+
+class Silence(commands.Cog):
+    """Commands for stopping channel messages for `verified` role in a channel."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.muted_channels = set()
+        self.bot.loop.create_task(self._get_instance_vars())
+
+    async def _get_instance_vars(self) -> None:
```

Comment:
If commands are used before this completes, things will likely blow up. Use an `asyncio.Event` to have commands wait for this to complete.

Reply:
How necessary is this? The vars finish loading in milliseconds after the bot connects and the commands become available for invoking.

Reply:
`wait_until_guild_available` may take a long time. It'd be a rare occurrence but I think it's a low effort to make it more robust.